sycl/doc/extensions/proposed/sycl_ext_oneapi_append_and_shift.asciidoc

```diff
@@ -0,0 +1,128 @@
+While this feature can easily be reproduced manually, is indicated below in the section "possible implementation", an extension is feasible since the underlying `pass:[__]spirv_SubgroupShuffleDownINTEL` and `pass:[__]spirv_SubgroupShuffleUpINTEL` already take the proposed two values and support the required capabilities.
+Note that there are multiple alternatives for the API suggested below ("append_and_shift_group_*"). For example, 
+
+
+``` 
+template <typename Group, typename T> 
+shift_group_left(Group g, T c, Group::linear_id_type delta=1, T n = 0);
+```
+or
+
+```
+template <typename Group, typename T> 
+shift_group_left_app(Group g, T c, T n, Group::linear_id_type delta=1);
+```
+
+
+== Specification
+
+=== Feature test macro
+This extension provides a feature-test macro as described in the core SYCL specification. An implementation supporting this extension must predefine the macro `SYCL_EXT_ONEAPI_APPEND_AND_SHIFT` to one of the values defined in the table below. Applications can test for the existence of this macro to determine if the implementation supports this feature, or applications can test the macro's value to determine which of the extension's features the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+	
+|1
+|The APIs of this experimental extension are not versioned, so the feature-test macro always has this value.
+|===
+
+=== API
+
+```
+template <typename Group, typename T>
+T sycl::ext::oneapi::append_and_shift_group_left(Group g, T c, T n, Group::linear_id_type delta=1);
+```
+
+_Constraints_: Available only if `std::is_same_v<std::decay_t<Group>, sub_group>` is true and `T` is a trivially copyable type.
+
+_Preconditions_: `delta` must be the same value for all work-items in the group.
+
+_Returns_: The value `c` of the work-item with group local id `id+delta`. If `id+delta` is greater or equal than the group's linear size, the function returns `n` of the work-item with the group local id `id+delta-sg_size`.
+
+
+
+```
+template <typename Group, typename T>
+T sycl::ext::oneapi::append_and_shift_group_right(Group g, T c, T n, Group::linear_id_type delta=1);
+```
+_Constraints_: Available only if `std::is_same_v<std::decay_t<Group>, sub_group>` is true and `T` is a trivially copyable type.
+
+_Preconditions_: `delta` must be the same value for all work-items in the group.
+
+_Returns_: The value `c` of the work-item with group local id `id-delta`, if `id-delta` is greater or equal 0, and `n` of the work-item with group local id `sg_size+(id-delta)`, otherwise.
+
+
+=== Possible Implementation
+
+The feature can be implemented based on `pass:[__]spirv_SubgroupShuffleDownINTEL` and `pass:[__]spirv_SubgroupShuffleUpINTEL`. For devices without that capability, it can be implemented with the existing shuffle capabilities as follows:
+
+```

```

Comment:
You can enable syntax highlighting for these examples with:
```suggestion
```c++

```

Reply:
Fixed